stock_picking_auto_create_lot/models/stock_picking.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    def button_validate(self):
+        if self.picking_type_code == 'incoming':
```

Comment:
@sergio-teruel IMHO to be flexible, this should be an option in stock.picking.type (doing this you limit lot auto creation to incoming pickings).

So,

```suggestion
        if self.picking_type_id.auto_create_lot:
```